intuita-webview/src/main/Tree/index.tsx

```diff
@@ -1,41 +1,69 @@
 import ReactTreeView from 'react-treeview';
 import { ReactNode, memo, useState } from 'react';
 import { TreeNode } from '../../../../src/components/webview/webviewEvents';
+import { generateRandomColor } from '../../utilities';
 
 type Props = {
+	index: number;
 	depth: number;
 	node: TreeNode;
 	renderItem({
+		index,
 		node,
 		depth,
 		open,
 		setIsOpen,
+		color,
+		isLastChild,
 	}: {
+		index: number;
 		node: TreeNode;
 		depth: number;
 		open: boolean;
 		setIsOpen: (value: boolean) => void;
+		color: string;
+		isLastChild: boolean;
 	}): ReactNode;
+	color: string;
+	isLastChild: boolean;
 };
 
-const Tree = ({ node, depth = 0, renderItem }: Props) => {
+const Tree = ({
+	node,
+	depth = 0,
+	renderItem,
+	color: colorProp,
+	index,
+	isLastChild,
+}: Props) => {
+	const hasNoChildren = !node.children || node.children.length === 0;
+	const isFolderBreakdown = [
+		'folderElement',
+		'acceptedFolderElement',
+	].includes(node.kind);
 	const [open, setIsOpen] = useState(depth === 0);
+	const [color] = useState(hasNoChildren ? colorProp : generateRandomColor());
+	const treeItem = renderItem({
+		index,
+		isLastChild,
+		color,
+		node,
+		depth,
+		open,
+		setIsOpen,
+	});
 
-	const treeItem = renderItem({ node, depth, open, setIsOpen });
-
-	if (!node.children || node.children.length === 0) {
+	if (hasNoChildren) {
 		return <>{treeItem}</>;
 	}
 
 	const folderElements: TreeNode[] = [];
 	const caseByFolderElements: TreeNode[] = [];
 
-	// separate folder elements and caseByFolder elements
-	// since we want to display all caseByFolder elements above folder elements
-	if (
-		node.kind &&
-		['folderElement', 'acceptedFolderElement'].includes(node.kind)
-	) {
+	// separate folder children and caseByFolder children
+	// since we want to display all caseByFolder children at the current depth
+	// while we want to display all folder children at 1 level deeper depth
+	if (node.kind && isFolderBreakdown) {
```

Comment:
can `node.kind` be set to `null`?

Reply:
Noope, we can remove `node.kind`. Good catch!